src/_utils.ts

```diff
@@ -15,19 +27,29 @@ export const isSSL = /^https|wss/;
  *    common.setupOutgoing(outgoing, options, req)
  *    // => { host: ..., hostname: ...}
  *
- * @param {Object} Outgoing Base object to be filled with required properties
- * @param {Object} Options Config object passed to the proxy
- * @param {ClientRequest} Req Request Object
- * @param {String} Forward String to select forward or target
+ * @param outgoing Base object to be filled with required properties
+ * @param options Config object passed to the proxy
+ * @param req Request Object
+ * @param forward String to select forward or target
  *
- * @return {Object} Outgoing Object with all required properties set
+ * @return Outgoing Object with all required properties set
  *
  * @api private
  */
-export function setupOutgoing(outgoing, options, req, forward?) {
+export function setupOutgoing(
+  outgoing: httpNative.RequestOptions & httpsNative.RequestOptions,
+  options: ProxyServerOptions & {
+    target: NormalizedProxyTarget;
+    forward: NormalizedProxyTargetUrl;
+  },
+  req: httpNative.IncomingMessage,
+  forward?: "forward" | "target",
+): httpNative.RequestOptions | httpsNative.RequestOptions {
   outgoing.port =
     options[forward || "target"].port ||
-    (isSSL.test(options[forward || "target"].protocol) ? 443 : 80);
+    (isSSL.test(options[forward || "target"].protocol ?? "undefined")
```

Comment:
> `options[forward || "target"].protocol ?? "undefined"`

This looks strange, but this is to make types happy while keeping the behavior as before. Probably this can be
```suggestion
    (isSSL.test(options[forward || "target"].protocol ?? "http")
```